src/commands/nix/install.rs

```diff
@@ -0,0 +1,212 @@
+                        if line.contains(NIXOS_PRIORITY_MSG) || line.contains(NIX_PKG_PRIORITY_MSG)
+                        {
+                            // send to priority err and all err
+                            tx.send((None, Some(line.clone()), Some(line))).unwrap();
+                        } else {
+                            // send to all err
+                            // useful if the build fails for reasons other than priority
+                            tx.send((None, None, Some(line))).unwrap();
+                        }
+                    } else {
+                        tx.send((Some(line), None, None)).unwrap();
+                    }
+                }
+            }
+        });
+
+        while let Ok((line_opt, priority_err_opt, all_errs_opt)) = rx.recv() {
+            if let Some(line) = line_opt {
+                info!("hi: {line}");
+            }
+            if let Some(err) = priority_err_opt {
+                priority_err.push(err);
+            }
+            if let Some(err) = all_errs_opt {
+                all_errs.push(err);
+            }
+        }
+        handle.join().unwrap();
+    }
+}
+
+/// Given an iterator over a priority error message, get the priority for the installed packages
+/// and prioritize the newly installed package.
+///
+/// This does not incur an overhead since nix will check if the package is already installed.
+fn auto_prioritize_installed_package(
+    mut iter: SplitWhitespace,
+    msg_len: usize,
+    link: String,
+) -> Result<bool> {
+    for _ in 0..msg_len {
+        iter.next();
+    }
+    if let Some(given_priority) = iter.next() {
+        let chars = given_priority.chars();
+        if let Ok(current_pkg_priority) = chars
+            .filter(|c| c.is_ascii_digit())
+            .collect::<String>()
+            .parse::<i32>()
+        {
+            try_prioritize(current_pkg_priority, link)?
+        }
+    }
+
+    Ok(true)
+}
+
+/// `nix profile install --priority` can be negative, so here we just continue to try
+/// installing the package with decreasing priority number until the error goes away.
+///
+/// There currently isn't a way to check the priority of packages other than the error
+/// provided by nix when installing a package that it finds a conflict with.
+fn try_prioritize(mut pkg_priority: i32, link: String) -> Result<()> {
+    pkg_priority -= 1;
+    let output = Command::new(NIX_CMD)
+        .args(PROFILE_INSTALL_ARGS)
+        .arg(PRIORITY_FLAG)
+        .arg(pkg_priority.to_string())
+        .arg(link.clone())
+        .output()?;
+    if !output.stderr.is_empty() {
+        let stderr_str = String::from_utf8_lossy(&output.stderr);
+        if stderr_str.contains(NIXOS_PRIORITY_MSG) || stderr_str.contains(NIX_PKG_PRIORITY_MSG) {
+            // recursively decriment the package priority until the
+            // newly installed package has the highest priority
+            try_prioritize(pkg_priority, link)?
+        }
+    }
```

Comment:
Consider using the rust nix crate instead of the nix CLI: https://docs.rs/nix/latest/nix/